### PR TITLE
[FLINK-8776][flip6] Use correct port for job submission from Web UI

### PR DESCRIPTION
**Pull request by GJL**
## What is the purpose of the change

*Use correct port for job submission from Web UI when Flink is deployed in YARN session mode.*

cc: @tillrohrmann 

## Brief change log
  - *Use address of local WebMonitorEndpoint for the job submission from the Web UI.*
  - *Rename TestingLeaderRetrievalService to SettableLeaderRetrievalService and move class out of test directory.*

## Verifying this change

This change added tests and can be verified as follows:

  - *Added unit tests for SettableLeaderRetrievalService.*
  - *Manually submitted Job in YARN session.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (**yes** / no / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/leaderretrieval/SettableLeaderRetrievalService.java [hunk 1]
```diff
@@ -16,36 +16,34 @@
...
-	private volatile LeaderRetrievalListener listener;
+	private LeaderRetrievalListener listener;
 
-	public TestingLeaderRetrievalService() {
+	public SettableLeaderRetrievalService() {
 		this(null, null);
 	}
 
-	public TestingLeaderRetrievalService(String leaderAddress, UUID leaderSessionID) {
+	public SettableLeaderRetrievalService(String leaderAddress, UUID leaderSessionID) {
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/leaderretrieval/SettableLeaderRetrievalService.java]: `Nullable` annotations are missing

File: flink-runtime/src/test/java/org/apache/flink/runtime/leaderretrieval/SettableLeaderRetrievalServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for {@link SettableLeaderRetrievalService}.
+ */
+public class SettableLeaderRetrievalServiceTest {
```
Comment (tillrohrmann) [inline flink-runtime/src/test/java/org/apache/flink/runtime/leaderretrieval/SettableLeaderRetrievalServiceTest.java]: `TestLogger` missing.

Comment (GJL) [reply flink-runtime/src/test/java/org/apache/flink/runtime/leaderretrieval/SettableLeaderRetrievalServiceTest.java]: fixed

File: flink-runtime/src/main/java/org/apache/flink/runtime/leaderretrieval/SettableLeaderRetrievalService.java (hunk 1 above)
Comment (GJL) [reply flink-runtime/src/main/java/org/apache/flink/runtime/leaderretrieval/SettableLeaderRetrievalService.java]: fixed

**Pull request closed without merging**